modules/compute/vm-instance/variables.tf

```diff
@@ -73,6 +73,16 @@ variable "name_prefix" {
   default     = null
 }
 
+variable "use_deployment_name_in_resource_name" {
+  description = <<-EOT
```

Comment:
nit. Is it common practice for terraform to use ternary logic on optional bool? We could use "enum" (as in https://www.puppeteers.net/blog/terraform-enum-data-type/) for better interfacing:
`use_deployment_name_in_resource_name := "always" | "never" | "only_if_no_prefix"`

Reply:
I thought about enum. I strayed away because it is a bit clunker to define for user (having to get exact string vs setting bool), but given the rest of the critique, I will move away from bool in favor of enum if the three options are needed.

Reply:
I ended up picking a solution that did not need 3 options, so it is still a bool. Closing but please reopen if needed.